core/src/main/java/org/infinispan/Version.java

```diff
@@ -24,25 +26,39 @@
 
    private static final Version INSTANCE = new Version();
 
+   private final String version;
+   private final String codename;
    private final byte[] versionId;
    private final String moduleSlot;
    private final short versionShort;
    private final short marshallVersion;
    private final String majorMinor;
    private final String major;
 
+
    private Version() {
-      String parts[] = getParts(Injected.getVersion());
+      Properties properties = new Properties();
+      try (InputStream is = Version.class.getResourceAsStream("/META-INF/infinispan-version.properties")) {
```

Comment:
That was indeed an avenue I wanted to explore, but:
- the manifest is generated by the maven archive / bundle plugin only during the package goal (you could create a temporary manifest just for the sake of these)
- the testsuite relies on having the version available for a number of things (marshalling, configuration, etc)